### DISCUSSION TITLE
How to enable jmx options on zookeeper for datadog

### DISCUSSION THREAD

**Discussion by shanmukha511** [strimzi/strimzi-kafka-operator#6686] [Q&A]
_Please use this to only for bug reports. For questions or when you need help, you can use the [GitHub Discussions](https://github.com/strimzi/strimzi-kafka-operator/discussions), our [#strimzi Slack channel](https://slack.cncf.io/) or out [user mailing list](https://lists.cncf.io/g/cncf-strimzi-users/topics)._

**Describe the bug**
 i have added jmxoptions under zookeeper in kafka.yaml .But after deployed i was not able to see in process .I need to send zookeeper metrics to datadog.Is there any way to do it?
 
 **kafka.yaml**
 zookeeper:
      replicas: 3
      jmxOptions: {}

**Expected behavior**
it has to show jmx attributes wn we do ps -ef on zookeeper nodes.

**Environment (please complete the following information):**
 - Strimzi version: [e.g. main, 0.22.1]
 - Installation method: [e.g. YAML files, Helm chart, OperatorHub.io]
 - Kubernetes cluster: [e.g. Kubernetes 1.20, OpenShift 4.7]
 - Infrastructure: [e.g. Amazon EKS, Minikube]

**YAML files and logs**

Attach or copy paste the custom resources you used to deploy the Kafka cluster and the relevant YAMLs created by the Cluster Operator.
Attach or copy and paste also the relevant logs.

**Comment by scholzj** [comment]
TBH, I'm not really sure what are you asking for. The docs describe how can you enable JMX. The resource snippet you shared is not properly formatted, so it is hard to say if you have it correct. But JMX is a socket where you connect with the client. There is not much to see with `ps`. 

You should also double check if it is supported on your version - since you seem to use fairly old version of Strimzi and JMX support for ZooKeeper had been added only later - IIRC it was some time after the Kafka support.

**Reply by michael-barker** [reply]
@scholzj Do you know if there's an open issue around version 0.21.1 or earlier and enabling JMX? I believe this is an option I also have where we set `jmxOptions: {}` for our Kafka CustomResource but the JMX port isn't exposed.

**Reply by scholzj** [reply]
WDYM with open issue? If it is not supported there then nobody is going to add it to such an old version and do a new release.

**Reply by michael-barker** [reply]
I figured there probably wouldn't be a patch release but thought I'd ask if there's a known workaround or anything. I found [this discussion](https://github.com/strimzi/strimzi-kafka-operator/discussions/6318) as well and looks like I had upvoted it back then. We plan on upgrading but it keeps getting put off because of the removal of the old CRDs and migrating to the new ones.

**Reply by zamsong123** [reply]
Hi @scholzj I tried to find some strimzi document on monitoring with Datadog, but didn't find any clue, can you please help to advise is monitoring with datadog recommended by Strimzi? and if yes, is there any document on this (integration for Strimzi Kafka and Datadog)

**Reply by scholzj** [reply]
I do not have any experience with Datadog, sorry. The preferred way from Strimzi point of view is Prometheus monitoring. There are multiple discussions about Datadog, so maybe you can read through them?

**Reply by michael-barker** [reply]
This should be resolved by #7535.
